### ISSUE TITLE
SPMC 16.5.0: Refresh rate adjustment does not work on AFTV2 anymore

### ISSUE THREAD

**Issue by Bab72de** [koying/SPMC#540]
Hi.

I have the problem that the automatic adjustment of the refresh rate does not work anymore on my AFTV2 with the released version of SPMC 16.5.0. However, the previous RC from November 21 (SPMC-20161121-1c72918-spmc-jarvis-armeabi-v7a.apk) works flawlessly. In this version I could change the refresh rate of the GUI and automatic changes when starting a video worked too. In the current released version the option to change the GUI refresh rate is greyed out and also no change when playing a video.
This the log shows in the RC:
18:50:04.710 T:18446744072390808472  NOTICE:  fps: 23.976024, pwidth: 1478, pheight: 1080, dwidth: 1478, dheight: 1080
18:50:04.710 T:18446744072390808472  NOTICE: Display resolution ADJUST : 1920x1080 @ 24.000000 - Full Screen (18) (weight: 0.001)

And this in the released version:
18:43:15.982 T:18446744072408261472  NOTICE:  fps: 25.000000, pwidth: 1280, pheight: 720, dwidth: 1280, dheight: 720
18:43:15.982 T:18446744072408261472  NOTICE: Display resolution ADJUST : 1920x1080 @ 60.000004 - Full Screen (16) (weight: 0.200)

**Comment by coolllman** [comment]
On AFTV1 the same, 16.8.2 all works

**Comment by mrmojo666** [comment]
same on  philips tv with 16.5.0 refresh rate gui is grayed out and if i activate the automatic refresh change i get only 60 hz .... where i can find the apk you mentioned that is working ? i would like to try on my philips tv

**Comment by Bab72de** [comment]
The (for me) last working apk can be found here (in the first post): https://github.com/koying/SPMC/issues/435

**Comment by mrmojo666** [comment]
thanks, tonight i'm going to test it

**Comment by koying** [comment]
Could someone point me to a full log on AFTV2? The extract in OP doesn't make sense...
Re Philips and Sony TV, they don't support refresh rate adjustment at all.

**Comment by mrmojo666** [comment]
sorry for asking, why they shouldn't support refresh rate changes ? Until some months ago a lot of people was saing those tv doesn't support passthrough form arc, but in the end SPMC made to get it working and now the same happens in krypton ;)

**Comment by koying** [comment]
@mrmojo666 You're very welcome to ask, but as I doubt Sony/Philips read the SPMC issue tracker, you'd better ask them.

**Comment by classicjazz** [comment]
I updated the Wiki re Sony/Philips TVs.

**Comment by mrmojo666** [comment]
@koying I agree they doesn't care  about 3rd party software, but i'm not asking you to fix their tvs, just what  makes u think that  is their issue ? As i wrote before i read lot of message on kodi forum about those tv is not sopporting passthrough at all, but in fact you were able to make it working, that's why i'm asking ;)
btw, compliments for your awesome fork

**Comment by Starfiresg1** [comment]
This is a complete log taken from a AFTV1 connected to a Panasonic TV with debug logging enabled - launching the app and playing a video file with 24p:

[Pastebin](http://pastebin.com/2p86tHJx)

**Comment by koying** [comment]
First, they should (finally) upgrade to Marshmallow (api 23), then implement the Android API actually enabling refresh rate/resolution switching...

**Comment by Bab72de** [comment]
I can temporarily upgrade again and then send you the log.

**Comment by koying** [comment]
@Starfiresg1 Only1080p60 detected. I bet you forced it at the AFTV settings level.
You have to let it on "Auto" for SPMC to be able to switch

**Comment by Starfiresg1** [comment]
@koying It's currently set to "Auto" (I just checked) - don't think I ever even touched that

**Comment by koying** [comment]
Arf. Stupid last minute change.
Test build cooking

**Comment by Bab72de** [comment]
Here are two log files, one the version from November 21 and one the released 16.5.

[logs.zip](https://github.com/koying/SPMC/files/627713/logs.zip)

**Comment by koying** [comment]
Please test http://download.semperpax.com/spmc/android-arm/SPMC-20161202-0ea8628-16.5.1-armeabi-v7a.apk

**Comment by Bab72de** [comment]
Refresh rate change works again. 
Thank you very much, koying.

**Comment by Starfiresg1** [comment]
I can confirm this too on AFTV1 - log file shows several resolutions at startup and switching of the display rate upon launching a video file works - great work and thanks for the quick fix.

**Comment by ttttubby** [comment]
Any chance we can get a launcher version of the apk?  Or update the official build?
